app/server/appsmith-git/src/main/java/com/appsmith/git/helpers/FileUtilsImpl.java

```
@@ -160,15 +167,40 @@ public Mono<Path> saveApplicationToGitRepo(Path baseRepoSuffix,
                     saveFile(applicationGitReference.getTheme(), baseRepo.resolve(CommonConstants.THEME + CommonConstants.JSON_EXTENSION), gson);
 
                     Path pageDirectory = baseRepo.resolve(PAGE_DIRECTORY);
+
                     try {
                         // Remove relevant directories to avoid any stale files
-                        FileUtils.deleteDirectory(pageDirectory.toFile());
-                        FileUtils.deleteDirectory(baseRepo.resolve(ACTION_DIRECTORY).toFile());
-                        FileUtils.deleteDirectory(baseRepo.resolve(ACTION_COLLECTION_DIRECTORY).toFile());
+                        Mono<List<Boolean>> pageList = Flux.fromStream(Files.walk(pageDirectory)
+                                .map(Path::toFile))
+                                .map(File::delete)
+                                .collectList();
+
+                        Mono<List<Boolean>> actionList = Flux.fromStream(Files.walk(baseRepo.resolve(ACTION_DIRECTORY)))
+                                .map(Path::toFile)
+                                .map(File::delete)
+                                .collectList();
+
+                        Mono<List<Boolean>> actionCollectionList = Flux.fromStream(Files.walk(baseRepo.resolve(ACTION_COLLECTION_DIRECTORY)))
+                                .map(Path::toFile)
+                                .map(File::delete)
+                                .collectList();
+                        return Mono.zip(pageList, actionList, actionCollectionList)
+                                .flatMap(objects -> Mono.zip(Mono.just(baseRepo), Mono.just(gson), Mono.just(validFileNames)));
+
                     } catch (IOException e) {
                         log.debug("Unable to delete directory for path {} with message {}", pageDirectory, e.getMessage());
+                        //return Mono.error(new IOException("Unable to delete directory for path {} with message {}"));
```

Do we want to continue even if the delete operation failed for some reason? Asking this because this may result in resource being deleted in DB but still present in file system. IMO we should throw the exception as earlier

---

There was no error thrown before. It was not at all present. I have added this to test few things.

---

This may happen when we don't have permission to perform the file ops. If you think this will not happen then we can keep this in logs for now